### PR TITLE
Fixed various Message.Finish() invocations

### PR DESCRIPTION
**Pull request by slinkydeveloper**
The integration tests that uses `protocol/test.SendReceive` now correctly test this behaviour

Signed-off-by: Francesco Guardiani <francescoguard@gmail.com>

### CODE REVIEW

File: v2/protocol/amqp/sender.go [hunk 1]
```diff
@@ -28,7 +28,8 @@ func (s *sender) Send(ctx context.Context, in binding.Message) error {
 		return err
 	}
 
-	return s.amqp.Send(ctx, &amqpMessage)
+	err = s.amqp.Send(ctx, &amqpMessage)
```
Comment (n3wscott) [inline v2/protocol/amqp/sender.go]: why this change?

Comment (slinkydeveloper) [reply v2/protocol/amqp/sender.go]: if error happened during `Send`, the `err` is used by the defer call

Comment (n3wscott) [reply v2/protocol/amqp/sender.go]: oooh to access inside the Finish.

Comment (n3wscott) [reply v2/protocol/amqp/sender.go]: line 22 needs the same change.
